lib/from_honeybee/_openapi/HoneybeeMapper.rb

```diff
@@ -0,0 +1,127 @@
+# this list of conditions and the following disclaimer.
+#
+# (2) Redistributions in binary form must reproduce the above copyright notice,
+# this list of conditions and the following disclaimer in the documentation
+# and/or other materials provided with the distribution.
+#
+# (3) Neither the name of the copyright holder nor the names of any contributors
+# may be used to endorse or promote products derived from this software without
+# specific prior written permission from the respective party.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE
+# UNITED STATES GOVERNMENT, OR THE UNITED STATES DEPARTMENT OF ENERGY, NOR ANY OF
+# THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
+# OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *******************************************************************************
+
+require 'urbanopt/scenario'
+require 'openstudio/common_measures'
+require 'openstudio/model_articulation'
+
+require 'json'
+
+module URBANopt
+    module Scenario
+      class HoneybeeMapper < SimulationMapperBase
+
+      # class level variables
+      @@instance_lock = Mutex.new
+      @@osw = nil
+      @@geometry = nil
+
+      def initialize()
+
+        # do initialization of class variables in thread safe way
+        @@instance_lock.synchronize do
+            if @@osw.nil?
+  
+              # load the OSW for this class
+              osw_path = File.join(File.dirname(__FILE__), 'honeybee_workflow.osw')
+              File.open(osw_path, 'r') do |file|
+                @@osw = JSON.parse(file.read, symbolize_names: true)
+              end
+
+            # configures OSW with extension gem paths for measures and files, all extension gems must be
+            # required before this
+            @@osw = OpenStudio::Extension.configure_osw(@@osw)
+            end
+          end
+      end
+
+      def create_osw(scenario, features, feature_names)
+
+        if features.size != 1
+          raise "TestMapper1 currently cannot simulate more than one feature"
+        end
+        feature = features[0]
+        feature_id = feature.id
+        feature_type = feature.type
+        feature_name = feature.name
+        if feature_names.size == 1
+          feature_name = feature_names[0]
+        end
```

Comment:
What is the purpose of this `if` statement here? Can we remove it?

Reply:
I see that the same code is in the baseline mapper so it's fine if you want to leave it in when we merge. It might be good to know why it's there if you find out, though.

Reply:
This was added for a similar reason, to possible extend create_osw to take more than 1 feature. Currently it is passed in as an array but is taking just the first feature_name from it.